README.md

```diff
@@ -1,4 +1,90 @@
 # roadhog
 
 ## Description
-Library that is connected to redux, use to fetch api, and to dispatch saga event
\ No newline at end of file
+Library that is connected to redux, use to fetch api, and to dispatch saga event
+
+## Installation
+```
+npm install --save roadhog-saga
+```
+or
+```
+yarn add roadhog-saga
+```
+
+## Usage
+In one of your saga, you want to retrieve all `POSTS` from your API.
+
+You should :
+ 1. Configure your redux to set the route
+ 2. Write your saga
+
+### 1 - Redux configuration
+Add a new reducer with a default state to `state.config` :
+
+**config.js**
+```es6
+export default () => ({
+  api: {
+    POSTS: {
+      GET: '/api/posts',
+    },
+  },
+})
+/* eslint-enable global-require */
+```
+
+**store.js**
+```es6
+import config from './config'
+
+const store = createStore(
+  combineReducers({
+    config,
+    // your others reducers
+  }),
+  compose(
+    applyMiddleware(sagaMiddleware, middleware),
+  ),
+)
+```
+
+### 2 - Saga
+In this exemple, we retrieve posts when an action `ACTION` is catched, and then we print these posts.
+```es6
+import { call } from 'redux-saga/effects'
+import roadhog from 'roadhog-saga'
+
+export default function* () {
+  yield takeLatest('ACTION', function* () {
+    const practitioners = call(roadhog('GET_POSTS'))
+    console.log(practitioners)
+  }
+}
+```
+
+### 3 - Error handling
+`roadhog-saga` triggers some redux actions to help you handle API calls :
+ - a `START` action : before a fetch is triggered
+ - an `ERROR` action : when the fetch is on error
+ - a `END` action : when the fetch is done (on error or not)
+
+This actions are named after your resources, with the previous example these actions are :
```

Comment:
The name of the action is generated from your resources. Based on the previous example these actions are